packages/Comments/NodeDocBlock/DocBlockUpdater.php

```diff
@@ -53,7 +55,7 @@ public function updateRefactoredNodeWithPhpDocInfo(Node $node): void
             return;
         }
 
-        $node->setDocComment(new Doc((string) $phpDocNode));
+        $node->setDocComment(new Doc($this->phpDocParserPrinter->print($phpDocNode)));
```

Comment:
@TomasVotruba this is based on @ondrejmirtes suggestion at https://github.com/rectorphp/rector/issues/7959#issuecomment-1646006214

to print `int|string` instead of `(int | string)`

Reply:
The issue with it is on printing doctrine annotation:

```diff
- * @Table("Table_Name")
+ * @Table ("Table_Name")
```

which cause a space before `(`

Reply:
Removing space on `@Table` seems can be handled by set `parseDoctrineAnnotations` arg to true:

https://github.com/rectorphp/rector-src/blob/729f03b6e9abccb7c4d7a4515e4dfed61e725e71/packages/BetterPhpDocParser/PhpDocParser/BetterPhpDocParser.php#L57-L58

but that will cause invalid FQCN identifiier:

```diff
 /**
- * @Table("Table_Name")
+ * @Doctrine\ORM\Mapping\Table("Table_Name")
```

that may need another tweak for doctrine use case

Reply:
it seems space is from custom `SpacelessPhpDocTagNode` node. and it got spaced at phpdoc-parser printer:

https://github.com/phpstan/phpdoc-parser/blob/4a1ab8e11e9957f9cc9f89f87a7c912489f08119/src/Printer/Printer.php#L202

Reply:
phpdoc-parser now has full Doctrine annotation support. You can remove a lot of hacks on your side thanks to it.

Reply:
@ondrejmirtes the current issue is FQCN-ize the identifier that used on rector to get resolved class if that `parseDoctrineAnnotations` be enabled https://github.com/rectorphp/rector-src/pull/4573#discussion_r1274802535

Reply:
@ondrejmirtes I created new PR to phpdoc-parser :

- https://github.com/phpstan/phpdoc-parser/pull/206 

to add new node: `SpacelessPhpDocTagNode`